src/python/espressomd/particle_data.pyx

```diff
@@ -1896,6 +1898,128 @@ cdef class ParticleList(object):
+            Sets the particle type for the added particles. If there
+            is a topology file give that contains a types for the
+            particles, the particles get types by the order in the
+            topology file plus firstype.  If the corresponding type in
+            the topology file has a charge, it is used, otherwise the
+            particle charge defaults to zero.
+        'first_id': integer
+            The particles get consecutive id’s in the order of the pdb
+            file, starting at firstid. Please be aware that existing
+            particles get overwritten by values from the file.
+        'itp_file': string, optional (default: empty)
+            filename of the ITP source
+        'first_type': integer, optional (default: 0)
+            Different species can be assigned different types.  The
+            types are numbered ascending from this value.
+        'lj_rel_cutoff': float, optional (default: 2.5)
+            The cutoff is determined by lj_rel_cutoff as
+            :math:`\\mathtt{lj\\_rel\\_cutoff}\\times\\sigma_{ij}` in a
+            relative fashion. The potential is shifted so that it
+            vanishes at the cutoff.
+        'rescale_box': bool, optional (default: False)
+            If this option is given, the box size if increased to hold
+            the particles if necessary. If it is not set and the
+            particles do not fit into the box, the behavior is
+            undefined.
+        'lj_internal': bool, optional (default: False)
+            *not documented*
+        'lj_diagonal': bool, optional, implies lj_internal (default: False)
+            *not documented*
+        'lj_with': list, optional (default: empty)
+            lj_with is a list of the form::
+
+                [othertype1, epsilon1, sigma1, othertype2, epsilon2, sigma2, ..., othertypeN, epsilonN, sigmaN].
+
+            The lj_with section produces Lennard-Jones interactions
+            between the type othertype and the types defined by the
+            topology file. The interaction parameters are calculated
+            as :math:`\\epsilon_{\\mathrm{othertype},j} =
+            \\sqrt{\\epsilon_{\\mathrm{othertype}} \\epsilon_j}` and
+            :math:`\\sigma_{\\mathrm{othertype},j} =
+            \\frac{1}{2}(\\sigma_{\\mathrm{othertype}} + \\sigma_j)`
+            where :math:`j` runs over the atomtypes defined in the
+            topology file. This corresponds to the combination rule 2
+            of Gromacs.
+
+        Examples
+        --------
+
+        >>> import espressomd
+        >>> system = espressomd.System()
+        >>> system.part.(pdb_file="ortho_dimer.pdb", type=0, first_id=0)
+
+        """
+
+        check_type_or_throw_except(pdb_file     , 1, str  , "pdb_file must be string")
+        check_type_or_throw_except(itp_file     , 1, str  , "itp_file must be string")
+        check_type_or_throw_except(type         , 1, int  , "type must be an integer")
```

Comment:
`type` is a bad name for a Python variable as it's a keyword. You can overwrite the keyword like you've done, but that can be confusing at times.

Reply:
True, but I didn't want to break compatibility in terms of naming.  You can also easily restore the original meaning by simply doing `del type`.

Reply:
We want to be consistent, right?
https://github.com/espressomd/espresso/blob/db8b559b756dd1617021df88364a9b3f3928f091/src/python/espressomd/particle_data.pyx#L85-L88

Reply:
Isn't this a new function? So you can choose the signature freely without compatibility concerns. I assume you did this to mimic the signature of `system.particle.add` -- though that does not have the overwriting problem because it only accepts a `type` argument through `**kwargs`.

Reply:
Properties named `type` don't have that problem because they don't pollute the local namespace with a `type` variable. They are only accessible through `self.type`. So if you want to name it `type` for consistency, you should go through `**kwargs`.